docs/operator-manual/applicationset/Generators-Plugin.md

```diff
@@ -0,0 +1,237 @@
+# Plugin Generator
+
+Plugins allow you to provide your own generator.
+
+* You can write in any language
+* Simple: a plugin just responds to RPC HTTP requests.
+* You can use it in a sidecar, or standalone deployment.
+* You can get your plugin running today, no need to wait 3-5 months for review, approval, merge and an Argo software
+  release.
+* You can combine it with Matrix or Merge.
+
+## Simple example
+
+Using a generator plugin without combining it with Matrix or Merge.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: myplugin
+spec:
+  generators:
+    - plugin:
+        # The name of the plugin used
+        name: my-plugin
+        # Specify the configMap where the plugin configuration is located
+        configMapRef: my-plugin
```

Comment:
Is there any imaginable scenario where we'd want the CM to live in a different ns? If so, maybe we can do a `configMapRef: {name: my-plugin}` structure for now and add the namespace field later if it proves useful.

Reply:
We considered CM like secrets. They do change the behavior of Argo server and can reference Argo secrets. Thus, security concerns must apply

Reply:
In line with @scrocquesel , I believe that the configuration of the plugin should be the responsibility of the ArgoCD administrator and therefore should be in the same namespace as ArgoCD.

Reply:
I think it's a good compromise.